lib/core/utils/preload.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Validated the preload object
+ * @param {Object | boolean} preload configuration object or boolean passed via the options parameter to axe.run
+ * @return {boolean}
+ * @private
+ */
+function isValidPreloadObject(preload) {
+	return (
+		typeof preload === 'object' &&
+		preload.hasOwnProperty('assets') &&
+		Array.isArray(preload.assets)
+	);
+}
+
+/**
+ * Returns a boolean which decides if preload is configured
+ * @param {Object} options run configuration options (or defaults) passed via axe.run
+ * @return {boolean}
+ */
+axe.utils.shouldPreload = function shouldPreload(options) {
```

Comment:
Could simplify this:

```js
axe.utils.shouldPreload = options => options && isValidPreloadObject(options.preload)
```

Reply:
Why was my comment marked as resolved? You didn't change anything or respond.....?

Reply:
Simply because, I do not want to check if `preload` is a valid `object` until I am sure that it is an `object`. Hence a `boolean` check and an `undefined` check earlier.

Preload can take 3 values: `true`, `false` or `{ assets: ['assetKey' }` as defined in the [docs](https://github.com/dequelabs/axe-core/pull/958/files/027b3a183847bfcf1c68e69ea3d453c191126675#diff-8be2f642524623d5012c7b2cf4fa4eb9R466).

Doing this `options && isValidPreloadObject(options.preload)` ignores and overrides the boolean values due to `&&`.

Scenarios (with your suggestion):

| Preload Value | options | isValidPreloadObject | output | expected |
| ------------- |---------| -------------------- | ------ | -------- | 
| `options: undefined or null` | false | false | false | false |
| `options: { preload : true }`| true  | false | false | true (BOOM!) |
| `options: { preload : false }`| true  | false | false | false |
| `options: { preload : { assets: ['cssom'] } }`| true  | true | true | true |

I am sure there is a clever way to shorten things further, but this keeps it easy to follow.
Prefer to keep this for that reason. 

Appreciate the comments.